### ISSUE TITLE
Add E2E tests for new update logic in Rancher Desktop

### ISSUE THREAD

**Issue by adamkpickering** [rancher-sandbox/rancher-desktop#4592]
We now have acceptance criteria in the epic for this issue (#3925). This issue is about creating E2E tests that satisfy these criteria.

I believe that tests of the same sort already exist for the changes to Upgrade Responder, but it wouldn't hurt to double check this.

**Comment by adamkpickering** [comment]
Closing this issue because, as far as I can tell, it is not possible to write playwright tests. This is because of our use of the `electron-updater` library: we cannot pass a different `currentVersion` to the library without breaking other things. I recommend replacement of this library with something that is easier to test with, and more understandable in general.

BATS tests do not work for this case either. There would need to be an automated way to:
1. Download and install different versions of Rancher Desktop in the middle of the test
2. Mock the Upgrade Responder server
3. Mock Github

And maybe more.

Basically, this is a hard thing to test in an automated fashion.
